pkg/securityscan/scanHandler.go

```diff
@@ -100,7 +100,12 @@ func (c *Controller) handleClusterScans(ctx context.Context) error {
 				//launch new on demand scan
 				c.mu.Lock()
 				logrus.Infof("Launching a new on demand Job for scan %v to run cis using profile %v", obj.Name, profile.Name)
-				configmaps, err := ciscore.NewConfigMaps(obj, profile, c.Name, c.ImageConfig)
+				benchmark, err := c.getClusterScanBenchmark(profile)
+				if err != nil {
+					v1.ClusterScanConditionReconciling.True(obj)
+					return objects, obj.Status, fmt.Errorf("Error when getting Benchmark: %v", err)
```

Comment:
This return is missing an unlock, but in general I think a defer should be used instead of trying to keep track of the lock. Looking at the code after the lock I don't see network calls so the defer should be safe here.